feat(bottomnavigationbar) persist badge on pressed/selected

lochstar opened the pull request:
Currently the badge is removed when the tab is pressed or selected.

I suggest we leave it up to the user to determine when to clear the badge. There's the case where the badge is showing a notification count that may change when reading/clearing notifications. Without this, we have to repeatedly show the badge during navigation.

farfromrefug commented:
@lochstar i agree but it is a breaking change. I suggest to add a `autoClearBadge` property and default it to `true` to prevent breaking change for users.

lochstar commented:
@farfromrefug I've added the `autoClearBadge` property.

I think it makes sense to have a `badge` property that can be passed to the component via the template, instead of using `showBadge()` and `removeBadge()`.

`badge` could be set to an empty string to display a badge without any text and `null/false` to remove the badge. Or we split it into two properties: `showBadge` and `badgeText`.

`showBadge()` and `removeBadge()` also currently only accept numbers, but the user may want to instead pass a string e.g. "99+"

Thoughts?

farfromrefug commented:
@lochstar having a badge property seems nice good too we can add that. 
Dont empty string make sense? like empty red badge for example? (if it works ...)
So i would say:
* add `badge` which accepts string/number
* add support for string to showBadge/removeBadge

good with you?

lochstar commented:
@farfromrefug I'm having a bit of trouble getting the demo apps to run.

Can we merge this as is and I will create a separate PR for the badge property. Turns out only iOS supports stings, so I will keep it as a number only.
